pytorch_lightning/utilities/apply_func.py

```diff
@@ -144,7 +144,7 @@ def apply_to_collection(
                 )
             if not field_init or (not include_none and v is None):  # retain old value
                 v = getattr(data, field_name)
-            setattr(result, field_name, v)
+            object.__setattr__(result, field_name, v)
```

Comment:
I don't think we should use this all the time. Only fall back to this if necessary and use `setattr(result, field_name, v)` the remaining time.